semtypes/src/main/java/io/ballerina/types/typeops/ListProj.java

```diff
@@ -113,32 +118,30 @@ static SemType listProjPath(Context cx, SubtypeData k, Conjunction pos, Conjunct
                     Atom d = p.atom;
                     p = p.next;
                     lt = cx.listAtomType(d);
-                    TwoTuple intersected = listIntersectWith(members, rest,
-                            lt.members, lt.rest);
+                    TwoTuple intersected = listIntersectWith(cx.env, members, rest, lt.members(), lt.rest());
                     if (intersected == null) {
                         return NEVER;
                     }
                     members = (FixedLengthArray) intersected.item1;
-                    rest = (SemType) intersected.item2;
+                    rest = (CellSemType) intersected.item2;
                 }
             }
             if (fixedArrayAnyEmpty(cx, members)) {
                 return NEVER;
             }
             // Ensure that we can use isNever on rest in listInhabited
-            if (rest != NEVER && isEmpty(cx, rest)) {
-                rest = NEVER;
+            if (cellInnerVal(rest) != NEVER && isEmpty(cx, rest)) {
```

Comment:
Can use `!Core.isNever()` as we did in other places

Reply:
Fixed